extension/extensionlimiter/limiterhelper/consumer.go

```diff
@@ -0,0 +1,313 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package limiterhelper // import "go.opentelemetry.io/collector/extension/extensionlimiter/limiterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/extension/extensionlimiter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Consumer is a builder for creating wrapped consumers with resource limiters
+//
+// This supports limiting by request_count, request_items, and
+// memory_size weight keys.
+//
+// The network_bytes weight key not supported because that information
+// is not available from the pdata object.
+type Consumer struct {
+	requestItemsLimiter extensionlimiter.ResourceLimiter
+	memorySizeLimiter   extensionlimiter.ResourceLimiter
+	requestCountLimiter extensionlimiter.ResourceLimiter
+}
+
+// Config stores configuration from Options.
+type Config struct {
+	requestItemsLimiter bool
+	memorySizeLimiter   bool
+	requestCountLimiter bool
+}
+
+// Option represents the consumer options
+type Option func(*Config)
+
+// WithRequestCountLimit configures the consumer to limit based on request count.
+func WithRequestCountLimit() Option {
+	return func(c *Config) {
+		c.requestCountLimiter = true
+	}
+}
+
+// WithRequestItemsLimit configures the consumer to limit based on item counts.
+func WithRequestItemsLimit() Option {
+	return func(c *Config) {
+		c.requestItemsLimiter = true
+	}
+}
+
+// WithMemorySizeLimit configures the consumer to limit based on memory size.
+func WithMemorySizeLimit() Option {
+	return func(c *Config) {
+		c.memorySizeLimiter = true
+	}
+}
+
+// NewConsumer creates a new limiterhelper Consumer
+func NewConsumer(provider extensionlimiter.Provider, options ...Option) *Consumer {
```

Comment:
Not sure you need the `Consumer` you can make:


```
func WrapTraces(provider extensionlimiter.Provider, nextConsumer consumer.Traces, options ...Option)
```

Reply:
Fixed.